src/App.js

```diff
@@ -18,27 +24,78 @@ const theme = createMuiTheme({
   }
 });
 
-export default function App() {
+const AuthRoute = props => {
+  if (!props.user.roles.includes('UNAUTHORIZED')) return <Redirect to="/" />;
+  return <Route {...props} />;
+};
 
+const App = () => {
   const [errors, setErrors] = useState({});
-  const [user, setUser] = useState({});
+  const [user, setUser] = useState({ roles: ['UNAUTHORIZED'] });
+  const [loginError, setLoginError] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const classes = useStyles();
+
+  useEffect(() => {
+    axios.defaults.headers.common = {'Authorization': localStorage.getItem('token')}
+    getUserProfile();
+  }, [])
+
```

Comment:
Are we going to keep all of the server communication logic in the App.js? I'm not used to doing this without redux so it might be fine but just looks odd at first for me, newbie.

Reply:
I don't think so that we should have all server communication i App.js, I just added here user logic, because for me it looked logical, but you can see that in profile page I have also server communication when changing password. So I don't think so that all server communication should be in App.js